Modules/_winapi.c

```diff
@@ -774,12 +774,144 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject* sortenvironmentkey(PyObject *module, PyObject *item) {
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    ""
+};
+
+static PyObject *
+normalize_environment(PyObject* environment)
+{
+    PyObject *result = NULL, *keys = NULL, *keyfunc = NULL, *sort = NULL,
+             *args = NULL, *kwargs = NULL;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
```

Comment:
```suggestion
    }
    asssert(PyList_CheckExact(keys));
```
The `PyMapping_Keys` returns an exact list. This assertion is to make sure this is ineed the case.

Reply:
You mean remove the `if (keys == NULL) { ... }` and changed to using the `asssert(PyList_CheckExact(keys))`?

Reply:
No, keep the `if (keys == NULL) { ...` and just add the assert. The assert will not be used in normal compilation, but in debug mode it will be compiled. In debug mode the assert will always succeed, unless someone changes the implementation of `PyMapping_Keys` such that it does not return an exact list any longer.
(chances of that happening are quite low, `PyMapping_Keys` is in the public API, so you can also skip the assert)

Reply:
If `PyMapping_Keys`'s [document](https://docs.python.org/3.9/c-api/mapping.html#c.PyMapping_Keys) guarantees it will returns a list, I think this assert is not necessary?